### PR TITLE
Problem: transfer private key can be compromised by leaked private view key

### PR DESCRIPTION
**Pull request by yihuang**
Solution: use different account_index(2) for view key derivation.

### CODE REVIEW

File: client-core/src/service/hd_key_service.rs [hunk 1]
```diff
@@ -16,9 +16,40 @@ const KEYSPACE: &str = "core_hd_key";
 struct HdKey {
     staking_index: u32,
     transfer_index: u32,
+    // curently only one viewkey per wallet, but it's good to keep it for uniformity.
+    viewkey_index: u32,
```
Comment (leejw51crypto) [inline client-core/src/service/hd_key_service.rs]: because transfer_index & viewkey_index always the same,
how about use just one index?

Comment (lezzokafka) [reply client-core/src/service/hd_key_service.rs]: ` staking_index `and  `transfer_index` are supposed to be the index for Staking address and Transfer address?

Comment (leejw51crypto) [reply client-core/src/service/hd_key_service.rs]: yes,
so it's policy issue.
for ease of restoration addresses when you resotre hdwallet.

which do you think easier?

Comment (yihuang) [reply client-core/src/service/hd_key_service.rs]: Staking addresses and transfer addresses are independent, so `staking_index` and `transfer_index` will have different values.

**Pull request merged**